packages/mds-transaction-service/service/provider.ts

```diff
@@ -28,11 +29,17 @@ import {
 } from './validators'
 
 export const TransactionServiceProvider: ServiceProvider<TransactionService> & ProcessController = {
-  start: TransactionRepository.initialize,
-  stop: TransactionRepository.shutdown,
+  start: async () => {
+    await Promise.all([TransactionRepository.initialize(), TransactionStreamKafka.initialize()])
+  },
+  stop: async () => {
+    await Promise.all([TransactionRepository.shutdown(), TransactionStreamKafka.shutdown()])
+  },
   createTransaction: async transaction => {
     try {
-      return ServiceResult(await TransactionRepository.createTransaction(validateTransactionDomainModel(transaction)))
+      const newTransaction = await TransactionRepository.createTransaction(validateTransactionDomainModel(transaction))
+      await TransactionStreamKafka.write(newTransaction)
```

Comment:
What should we do if postgres save succeeds but kafka write fails?

Currently: fail, which is misleading because the source of truth actually does have our transaction
Alternatives:
* Retry inline
* Periodically sweep through kafka vs postgres for missing transactions
* ???

Do we want to ignore this, create a followup spike, ...

Reply:
This is what it looks like in `mds-collector-backend`

```typescript
      // Write to Postgres
      const result = await CollectorRepository.insertCollectorMessages(
        messages.map(message => ({ schema_id, provider_id, message })),
        producer
          ? {
              // Write to Kafka prior to committing transaction
              beforeCommit: async () => {
                try {
                  await producer.write(messages)
                } catch (error) {
                  throw new ServerError('Error writing to Kafka stream', error)
                }
              }
            }
          : {}
      )

```

Reply:
I think this is probably what we should go with, only commit if the write to kafka succeeds.

Reply:
I like this pattern - commit-or-not

Reply:
Is committing if and only if the write to kafka succeeds still the plan?

Reply:
For transactions, yes @janedotx (all transaction sources can be retried later)